### PR TITLE
Atmega328p

### PR DESCRIPTION
**Pull request by AHorneffer**
I finally cleaned up a branch with the ATmega328p code. 
I haven't tested if my changes work on the other configurations. I tried to limit my changes to only affect compilation for the ATmega328p, but some testing is still called for.

### CODE REVIEW

File: firmware/main.c [hunk 1]
```diff
@@ -157,36 +166,44 @@ static void writeWordToPageBuffer(uint16_t data) {
...
   boot_page_fill(currentAddress.w, data);
   currentAddress.w += 2;
 }
 
 /* ------------------------------------------------------------------------ */
 static uint8_t usbFunctionSetup(uint8_t data[8]) {
   usbRequest_t *rq = (void *)data;
- 
+  
+  idlePolls.b[1]=0; // reset idle polls when we get usb traffic
```
Comment (cpldcpu) [inline firmware/main.c]: I noticed you moved around idlePolls resetting. What is the purpose of this?

Comment (AHorneffer) [reply firmware/main.c]: [searching, it's been a while...]

From my post on #132 on Nov 20, 2017:
> The other issue was that apparently the idle timer was reset on any USB communication, not only on communication with the command-line tool. So it got reset every time the OS was bugging the device with something

IIRC it kept the bootloader from starting the application program for as long as an active USB host is around.

**Comment by cpldcpu** [comment]
Thanks, a lot - merging and increasing version.

**Comment by AHorneffer** [comment]
Before you tag a new release: I just got an ATmega168p.
It shouldn't be that hard to get it to work on that one, too.

**Comment by cpldcpu** [comment]
Yeah, I will let this one sit for a while. I just had to advance the version because the previous state was already released as 2.03

**Pull request merged**